src/EnergyPlus/DataEnvironment.cc

```diff
@@ -181,6 +181,122 @@ namespace DataEnvironment {
 
 	// Functions
 
+	// Clears the global data in DataEnvironment.
+	// Needed for unit tests, should not be normally called.
+	void
+	clear_state()
```

Comment:
I'm in favor of that view of the world.  Data gets classified as time vary or not time varying.  Class are instantiated with all not time varying data, possibly from idf parser.  time varying data must be passed into and out of member functions.